### ISSUE TITLE
Add jackknife estimation of RSD multipoles

### ISSUE THREAD

**Issue by aphearin** [astropy/halotools#753]
As requested by @sssaito to support work on ongoing project with @duncandc - we will need jackknife error estimates on the tpcf multipole calculations, which are currently only supported for `tpcf` in 3d. 

I *think* that the current weighted-pair-counting scheme used in `tpcf` should carry over directly, but @duncandc should weigh in about this if there are subtleties that I'm not thinking of

**Comment by duncandc** [comment]
Yeah, we can use the same set-up in the tpcf_jackknife function.  This has long been on my list for both tpcf_s_mu_jackknife and tpcf_rp_pi_jacknife.  We should consider if it makes sense to combine these somehow with their non-jackknife cousins.  

I have been thinking about how we might pre-compute some results for large numbers of randoms to speed up these jackknife calculations.

**Comment by aphearin** [comment]
Yeah, #234 is also long-standing. Probably all these different variations would best done in a sprint, since they're all so similar. 

Re: combining things. That'd be fine, like with a keyword to the primary function(s) for whether or not to return jackknife covariances. But all the guts could remain the same. The keyword could just redirect control flow to existing functions. 

Re pre-compute. For gg-lensing chains I have been working on, I have been using the API of [delta_sigma_from_precomputed_pairs](http://halotools.readthedocs.io/en/latest/api/halotools.mock_observables.delta_sigma_from_precomputed_pairs.html) finding it to be pretty convenient, in case you do tackle this and a specific example would be helpful, either for halotools or otherwise.
